certbot/plugins/dns_common.py

```diff
@@ -29,13 +54,68 @@ def __init__(self, config, name):
 
         self._attempt_cleanup = False
 
+    @classmethod
+    def inject_parser_options(cls, parser, name):
+        """Set up argument parsing for this DNS plugin.
+
+        This is called by the plugin management framework to set up
+        the argument parser.
+
+        :param ArgumentParser parser: (Almost) top-level CLI parser.
+        :param str name: Unique plugin name.
+        """
+
+        # The inherited method will call add_parser_arguments to set up
+        # the parser of each argument.
+
+        super(DNSAuthenticator, cls).inject_parser_options(parser, name)
+
+        # Create an additional entry in the parser namespace for the
+        # override-challenge map. This is an initiallyt empty dict associating
+        # each requested certificate domain with the corresponding challenge
+        # override (see help for --dns-<plugin>-override-challenge).
+
+        # Use set_default to provide this dict as an entry in the namespace
+        # object returned by the parser. Note that set_default uses keyword
+        # arguments to determine the name of the entry to be set, and
+        # here the name is dynamically computed (depending on the plugin
+        # name), which is why we need to use ** syntax.
+
+        challenge_map_opt = common.dest_namespace(name) + "override_challenge_map"
+        parser.set_defaults(**{challenge_map_opt: {}})  # pylint: disable=star-args
+
+
     @classmethod
     def add_parser_arguments(cls, add, default_propagation_seconds=10):  # pylint: disable=arguments-differ
         add('propagation-seconds',
             default=default_propagation_seconds,
             type=int,
-            help='The number of seconds to wait for DNS to propagate before asking the ACME server '
-                 'to verify the DNS record.')
+            help='The number of seconds to wait for DNS to propagate before '
+                 'asking the ACME server to verify the DNS record.')
+        add('override-challenge',
```

Comment:
What do people (e.g. @adferrand, @joohoi, etc.) think of the name `validation-domain` rather than `override-challenge`. I personally think the former is a little more descriptive of what is being set here but I'm curious what other people think.

Reply:
I also think `validation_domain` is more explicit.

Reply:
Yes, agreed!

Reply:
I change all assets to validation-domain.